transcom/mymove
# MB-4949 TIO payment request queue pagination

## Description

> As a TIO, I want to see a subset of the Payment Request queue, so that I am not overwhelmed and my queue loads quickly.
> 
> Given that I am a TIO
>  When I log into MM and open the queue
>  Then I see a queue with 20 rows of Payment
>       Requests
>
> Given that I am a TIO
>  When I click on 'Prev', 'Next', or a
>       specific page from the drop down
>  Then the queue navigates accordingly
> 

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?
- After some thought and experimentation, I was able to make a re-usable component that passes in the queries hook with the expectation that the return value shape is always the same. Should reduce some boiler plate code regarding the queues or any table setup. The new component is called `TableQueue`.
- Added a new functionality for the seed data, making new payment requests.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

**To test**
1. Log in as TIO, http://officelocal:3000/devlocal-auth/login
2. Should now see pagination for the payment request queue

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4949) for this change

## Screenshots

![Nov-11-2020 14-25-50](https://user-images.githubusercontent.com/1522549/98871508-e342d880-2429-11eb-8c82-1379321cbfe4.gif)
